aikido_perception/src/AprilTagsModule.cpp

```diff
@@ -0,0 +1,170 @@
+/** 
+ * @file AprilTagsModule.cpp
+ * @author Shushman Choudhury
+ * @date Apr 14, 2016
+ * @brief The implementation of the April Tags detector. The function detectObjects
+ * waits for the next ROS message on the specified topic.
+ */
+
+#include <aikido/perception/AprilTagsModule.hpp>
+#include <aikido/perception/shape_conversions.hpp>
+#include "dart/common/Console.h"
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <ros/topic.h>
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <boost/make_shared.hpp>
+#include <Eigen/Geometry>
+#include <tf/LinearMath/Vector3.h>
+#include <tf/LinearMath/Quaternion.h>
+#include <stdexcept>
+
+namespace aikido{
+namespace perception{
+
+//================================================================================================================================
+
+AprilTagsModule::AprilTagsModule( ros::NodeHandle node, std::string markerTopic, std::shared_ptr<ConfigDataLoader> configData,
+								  const dart::common::ResourceRetrieverPtr& resourceRetriever,	
+								  std::string referenceFrameId, dart::dynamics::Frame* referenceLink):
+		mNode(node),
+		mMarkerTopic(markerTopic),
+		mConfigData(configData),
+		mResourceRetriever(resourceRetriever),
+		mReferenceFrameId(referenceFrameId),
+		mReferenceLink(referenceLink)
+{
+	mListener = new tf::TransformListener();
```

Comment:
See my comment in the header. Either store this by value or in a `unique_ptr`. It would also be nice to use the `TransformListener` that accepts a `NodeHandle` argument. Otherwise, the TF listener won't use the same `NodeHandle` as everything else. This has implications when using `CallbackQueue`s and `AsyncSpinner`s.